homeassistant/components/zha/sensor.py

```diff
@@ -7,20 +7,105 @@
 import logging
 
 from homeassistant.components.sensor import DOMAIN
-from homeassistant.const import TEMP_CELSIUS
+from homeassistant.const import STATE_UNKNOWN
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.util.temperature import convert as convert_temperature
-from .core import helpers
 from .core.const import (
-    DATA_ZHA, DATA_ZHA_DISPATCHERS, REPORT_CONFIG_MAX_INT,
-    REPORT_CONFIG_MIN_INT, REPORT_CONFIG_RPT_CHANGE, ZHA_DISCOVERY_NEW)
+    DATA_ZHA, DATA_ZHA_DISPATCHERS, ZHA_DISCOVERY_NEW, HUMIDITY, TEMPERATURE,
+    ILLUMINANCE, PRESSURE, METERING, ELECTRICAL_MEASUREMENT,
+    POWER_CONFIGURATION, GENERIC, SENSOR_TYPE, LISTENER_ATTRIBUTE,
+    LISTENER_ACTIVE_POWER, SIGNAL_ATTR_UPDATED, SIGNAL_STATE_ATTR)
 from .entity import ZhaEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 DEPENDENCIES = ['zha']
 
 
+# unit getters
+def get_temperature_unit(sensor):
+    """Get the configured temperature unit."""
+    # hass is bound eager on device and lazy on entity. Use device here
+    return sensor.zha_device.hass.config.units.temperature_unit
+
+
+# Formatter functions
+def pass_through_formatter(value, sensor):
+    """No op update function."""
+    return value
+
+
+def temperature_formatter(value, sensor):
```

Comment:
Do we need this? The base entity class already converts between F and C if it detects one of those as unit of measurement.

Reply:
will explore

Reply:
Done - still needs to process the raw result from the device but HA does the conversion naturally now.